tests/integration/test_active_learning.py

```diff
@@ -0,0 +1,150 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration tests for various forms of active learning implemented in Trieste.
+"""
+
+from __future__ import annotations
+
+import gpflow
+import numpy.testing as npt
+import pytest
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tests.util.misc import random_seed
+from trieste.acquisition.function import ExpectedFeasibility, PredictiveVariance
+from trieste.acquisition.rule import AcquisitionRule, EfficientGlobalOptimization
+from trieste.bayesian_optimizer import BayesianOptimizer
+from trieste.data import Dataset
+from trieste.models import TrainableProbabilisticModel
+from trieste.models.gpflow import GaussianProcessRegression
+from trieste.objectives import BRANIN_SEARCH_SPACE, branin, scaled_branin
+from trieste.objectives.utils import mk_observer
+from trieste.space import SearchSpace
+from trieste.types import TensorType
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, acquisition_rule",
+    [
+        (80, EfficientGlobalOptimization(PredictiveVariance())),
+    ],
+)
+def test_optimizer_learns_scaled_branin_function(
+    num_steps: int, acquisition_rule: AcquisitionRule[TensorType, SearchSpace]
+) -> None:
+    """
+    Ensure that the objective function is effecitvely learned, such that the final model
+    fits well and predictions are close to actual objective values.
+    """
+
+    search_space = BRANIN_SEARCH_SPACE
+
+    def build_model(data: Dataset) -> TrainableProbabilisticModel:
+        variance = tf.math.reduce_variance(data.observations)
+        kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+        prior_scale = tf.cast(1.0, dtype=tf.float64)
+        kernel.variance.prior = tfp.distributions.LogNormal(
+            tf.cast(-2.0, dtype=tf.float64), prior_scale
+        )
+        kernel.lengthscales.prior = tfp.distributions.LogNormal(
+            tf.math.log(kernel.lengthscales), prior_scale
+        )
+        gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+        gpflow.set_trainable(gpr.likelihood, False)
+
+        return GaussianProcessRegression(gpr)
+
+    num_initial_points = 20
+    initial_query_points = search_space.sample_halton(num_initial_points)
+    observer = mk_observer(scaled_branin)
+    initial_data = observer(initial_query_points)
+
+    model = build_model(initial_data)
+    final_model = (
+        BayesianOptimizer(observer, search_space)
+        .optimize(num_steps, initial_data, model, acquisition_rule)
+        .try_get_final_model()
+    )
+
+    test_query_points = search_space.sample_sobol(10000 * search_space.dimension)
+    test_data = observer(test_query_points)
+    predicted_means, _ = final_model.model.predict_f(test_query_points)  # type: ignore
+
+    npt.assert_allclose(predicted_means, test_data.observations, rtol=0.5)
```

Comment:
just for checking: allclose would still use the default atol, which is quite restrictive, right? otherwise, this is very loose?

Reply:
Also, this is not really consistent with the other test: the other tests a single average measure of accuracy, while this is a max. Max is fine, but maybe rewrite `crit = np.abs(predicted_means, test_data.observations)` and assert crit < something?

Reply:
yes, accuracy criteria is something to discuss, its difficult to say what to expect in terms of global accuracy with 100 observations, and we dont want to have tests that take ages - I'm open to suggestions here

Reply:
For this test, we probably want a criterion that would fail with `num_initial_points` and pass with more points, provided that they are not pathologically distributed (eg they should be roughly space-filling, not concentrated in one corner of the space). We can't do much more than that (eg we could see if we beat random search over replicated experiments, but we won't do that). But this is already a good test.

Reply:
Max error is probably better than average error because it will reward more space-filling designs.